WordPress/Classes/ViewRelated/Pages/ParentPageSettingsViewController.swift

```diff
@@ -0,0 +1,230 @@
+import UIKit
+
+
+private struct Section {
+    var rows: [Row]
+    var headerTitle: String
+
+    init(_ rows: [Row], headerTitle: String = "") {
+        self.rows = rows
+        self.headerTitle = headerTitle
+    }
+}
+
+
+private struct Row {
+    struct Constants {
+        static let identifier = "CheckmarkCellIdentifier"
+    }
+
+    enum RowType {
+        case topLevel
+        case child
+    }
+
+    var page: Page?
+    var type: RowType
+    var title: String {
+        switch type {
+        case .topLevel: return NSLocalizedString("Top level", comment: "Cell title for the Top Level option case")
+        case .child: return page?.postTitle ?? ""
+        }
+    }
+    var titleFont: UIFont {
+        switch type {
+        case .topLevel: return UIFont.systemFont(ofSize: 16.0)
+        case .child: return WPFontManager.notoRegularFont(ofSize: 17)
+        }
+    }
+}
+
+
+class ParentPageSettingsViewController: UIViewController {
+    @IBOutlet private var cancelButton: UIBarButtonItem!
+    @IBOutlet private var doneButton: UIBarButtonItem!
+    @IBOutlet private var tableView: UITableView!
+
+    private let postService = PostService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+    private var originalIndex: IndexPath?
+    private var selectedIndex: IndexPath? {
+        didSet {
+            doneButton.isEnabled = selectedIndex != originalIndex
+        }
+    }
+
+    private var sections: [Section]!
+    private var selectedPage: Page!
+    private var selectedParentId: NSNumber? {
+        guard let selectedIndex = selectedIndex else {
+            return nil
+        }
+        return sections[selectedIndex.section].rows[selectedIndex.row].page?.postID
+    }
+
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupUI()
+    }
+
+
+    func set(pages: [Page], for page: Page) {
+        selectedPage = page
+        originalIndex = originalIndexPath(with: pages)
+        selectedIndex = originalIndex
+
+        let rows = pages.map { Row(page: $0, type: .child) }
+        sections = [Section([Row(page: nil, type: .topLevel)]),
+                    Section(rows, headerTitle: NSLocalizedString("Pages", comment: "This is the section title"))]
+    }
+
+
+    // MARK: - Private methods
+
+    private func setupUI() {
+        WPFontManager.loadNotoFontFamily()
```

Comment:
that shouldn't be needed here :) if the fonts haven't loaded by the time this VC gets on screen, we probably have much bigger problems :D

Reply:
I checked around and we use `WPFontManager.loadNotoFontFamily()` within the `viewDidLoad` (Aztec and Share Extension Editor). I also checked this class and apparently is safe. If it doesn't finish to load the font while it calls the `WPFontManager.notoRegularFont(ofSize: 17)`, the class first check if the font is loaded. If not it tries to load it again. If there's something wrong it fallbacks to the system font.
By the way because it's a factory class I can move it before the VC is created. What do you think?

Reply:
>(Aztec and Share Extension Editor)

That makes sense though, because those can be launched outside of the context of the app (as you said, share extension!). This VC relies on the rest of the app having been set up, so the call to load fonts is superfluous here (and I think, actually close to a no-op given the implementation is wrapped in `dispatch_once`).